shop/src/test/java/ro/msg/learning/shop/StrategyUnitTest.java

```diff
@@ -0,0 +1,232 @@
+package ro.msg.learning.shop;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.context.annotation.Import;
+import ro.msg.learning.shop.configuration.StrategyUnitTestConfiguration;
+import ro.msg.learning.shop.exception.entity_exception.LocationException;
+import ro.msg.learning.shop.exception.entity_exception.ProductException;
+import ro.msg.learning.shop.model.*;
+import ro.msg.learning.shop.repository.*;
+import ro.msg.learning.shop.service.strategy_utils.MostAbundantStrategy;
+import ro.msg.learning.shop.service.strategy_utils.SingleLocationStrategy;
+import ro.msg.learning.shop.service.strategy_utils.StrategyService;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+@DataJpaTest
+@Import(StrategyUnitTestConfiguration.class)
+class StrategyUnitTest {
```

Comment:
Unit Tests should not work with the database. Should mock the data and the calls to the database. Please revisit this test entirely

Reply:
I've did just one test with mock, because I want to get a look over the security part. I will continue on testing the next days maybe

Reply:
StrategyUnitTest class will disappear in the end, as well as the OrderServiceTest class